src/main/java/com/jvm_bloggers/entities/newsletter_issue/NewsletterIssueRepository.java

```diff
@@ -18,4 +20,8 @@
 
     List<NewsletterIssue> findAllByOrderByPublishedDateDesc();
 
+    @Query("select case when count(*) > 0 then true else false end "
+        + "from NewsletterIssue where issueNumber = :issueNumber")
+    boolean issueNumberExist(@Param("issueNumber") Long issueNumber);
```

Comment:
@airborn unfortunately `exist(id)` method, as a parameter take PK of an entity and 'issueNumber' in our case is not PK. To remove this ugly SQL from repository code I see two options:

1) Use [Spring Data projections](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#projections) :
```
//NewsletterIssueId.java
public interface NewsletterIssueId {
    Long getId();
}

//NewsletterIssueRepository.java
Option<NewsletterIssueId> findIdByIssueNumber(Long issueNumber)
```

2. Introduce [QueryDsl](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#core.extensions.querydsl) to remove raw SQL queries but this is a new dependency and not sure if it is worth it just for one case.